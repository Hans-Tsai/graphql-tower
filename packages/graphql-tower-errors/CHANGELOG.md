# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="3.21.3"></a>
## [3.21.3](https://github.com/tmotx/graphql-tower/compare/v3.21.2...v3.21.3) (2017-12-05)




**Note:** Version bump only for package graphql-tower-errors

<a name="3.16.0"></a>
# [3.16.0](https://github.com/tmotx/graphql-tower/compare/v3.15.1...v3.16.0) (2017-11-19)


### Features

* **help:** added assertResult ([e6cd019](https://github.com/tmotx/graphql-tower/commit/e6cd019))




<a name="3.15.1"></a>
## [3.15.1](https://github.com/tmotx/graphql-tower/compare/v3.15.0...v3.15.1) (2017-11-18)


### Bug Fixes

* **error:** added NotAcceptableError ([7258d43](https://github.com/tmotx/graphql-tower/commit/7258d43))
