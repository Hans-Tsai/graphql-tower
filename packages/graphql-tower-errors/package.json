{
  "name": "graphql-tower-errors",
  "version": "1.0.0",
  "description": "",
  "main": "lib/index.js",
  "scripts": {
    "build": "rm -rf lib/* && babel src --ignore __tests__ --out-dir lib"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tmotx/graphql-tower.git"
  },
  "author": "tmotx",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tmotx/graphql-tower/issues"
  },
  "homepage": "https://github.com/tmotx/graphql-tower#readme",
  "files": [
    "lib"
  ],
  "peerDependencies": {
    "graphql": "^0.11.7"
  },
  "devDependencies": {
    "faker": "^4.1.0"
  }
}
