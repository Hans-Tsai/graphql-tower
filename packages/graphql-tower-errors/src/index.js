import { GraphQLError } from 'graphql/error';

export class ForbiddenError extends GraphQLError {
  symbol = 'ForbiddenError';

  constructor(message = 'ForbiddenError') {
    super(message);
  }
}

export class NotFoundError extends GraphQLError {
  symbol = 'NotFoundError';

  constructor(message = 'NotFoundError') {
    super(message);
  }
}

export class UnauthorizedError extends GraphQLError {
  symbol = 'UnauthorizedError';

  constructor(message = 'UnauthorizedError') {
    super(message);
  }
}

export class PaymentRequiredError extends GraphQLError {
  symbol = 'PaymentRequiredError';

  constructor(message = 'PaymentRequiredError') {
    super(message);
  }
}

export class NotAcceptableError extends GraphQLError {
  symbol = 'NotAcceptableError';

  constructor(message = 'NotAcceptableError') {
    super(message);
  }
}

export class GoneDataError extends GraphQLError {
  symbol = 'GoneDataError';

  constructor(message = 'GoneDataError') {
    super(message);
  }
}

export class ConflictError extends GraphQLError {
  symbol = 'ConflictError';

  constructor(message = 'ConflictError') {
    super(message);
  }
}

export class LockedError extends GraphQLError {
  symbol = 'LockedError';

  constructor(message = 'LockedError') {
    super(message);
  }
}

export class UnavailableForLegalReasonsError extends GraphQLError {
  symbol = 'UnavailableForLegalReasonsError';

  constructor(message = 'UnavailableForLegalReasonsError') {
    super(message);
  }
}
