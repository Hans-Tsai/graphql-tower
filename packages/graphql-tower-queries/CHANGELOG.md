# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="3.29.0"></a>
# [3.29.0](https://github.com/tmotx/graphql-tower/compare/v3.28.9...v3.29.0) (2017-12-23)


### Features

* **queries:** support totalCount on Connection ([aa9bc56](https://github.com/tmotx/graphql-tower/commit/aa9bc56))




<a name="3.26.0"></a>
# [3.26.0](https://github.com/tmotx/graphql-tower/compare/v3.25.2...v3.26.0) (2017-12-13)




**Note:** Version bump only for package graphql-tower-queries

<a name="3.24.2"></a>
## [3.24.2](https://github.com/tmotx/graphql-tower/compare/v3.24.1...v3.24.2) (2017-12-11)


### Bug Fixes

* **queries:** support data and contextValue on payload ([8cd7ef3](https://github.com/tmotx/graphql-tower/commit/8cd7ef3))




<a name="3.24.1"></a>
## [3.24.1](https://github.com/tmotx/graphql-tower/compare/v3.24.0...v3.24.1) (2017-12-10)


### Bug Fixes

* **redis-subscriptions:** added onMessage to support each params ([53d8c45](https://github.com/tmotx/graphql-tower/commit/53d8c45))




<a name="3.21.3"></a>
## [3.21.3](https://github.com/tmotx/graphql-tower/compare/v3.21.2...v3.21.3) (2017-12-05)




**Note:** Version bump only for package graphql-tower-queries

<a name="3.21.2"></a>
## [3.21.2](https://github.com/tmotx/graphql-tower/compare/v3.21.1...v3.21.2) (2017-12-01)


### Bug Fixes

* **queries:** run middleware on withFilter ([5e0a20b](https://github.com/tmotx/graphql-tower/commit/5e0a20b))




<a name="3.21.0"></a>
# [3.21.0](https://github.com/tmotx/graphql-tower/compare/v3.20.1...v3.21.0) (2017-11-30)


### Features

* **queries:** check offset from results ([01efee3](https://github.com/tmotx/graphql-tower/commit/01efee3))




<a name="3.20.0"></a>
# [3.20.0](https://github.com/tmotx/graphql-tower/compare/v3.19.10...v3.20.0) (2017-11-29)


### Features

* **queries:** added subscription ([b7498e0](https://github.com/tmotx/graphql-tower/commit/b7498e0))




<a name="3.12.0"></a>
# [3.12.0](https://github.com/tmotx/graphql-tower/compare/v3.11.0...v3.12.0) (2017-11-14)


### Features

* **queries:** support class name ([b515179](https://github.com/tmotx/graphql-tower/commit/b515179))




<a name="3.1.0"></a>
# [3.1.0](https://github.com/tmotx/graphql-tower/compare/v3.0.1...v3.1.0) (2017-10-26)


### Features

* **package:** setup lerna-semantic-release ([a4fb7d8](https://github.com/tmotx/graphql-tower/commit/a4fb7d8))
