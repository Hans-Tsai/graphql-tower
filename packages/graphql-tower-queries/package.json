{
  "name": "graphql-tower-queries",
  "version": "3.29.0",
  "description": "",
  "main": "lib/index.js",
  "scripts": {
    "build": "rm -rf lib/* && babel src --ignore __tests__ --out-dir lib"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tmotx/graphql-tower.git"
  },
  "author": "tmotx",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tmotx/graphql-tower/issues"
  },
  "homepage": "https://github.com/tmotx/graphql-tower#readme",
  "files": [
    "lib"
  ],
  "dependencies": {
    "graphql-tower-errors": "^3.21.3",
    "iterall": "^1.1.3",
    "lodash": "^4.17.4"
  },
  "peerDependencies": {
    "graphql": "^0.11.7"
  }
}
