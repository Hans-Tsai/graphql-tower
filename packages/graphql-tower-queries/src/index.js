export { default as Query } from './Query';
export { default as QueryWithNode } from './QueryWithNode';
export { default as QueryWithConnection } from './QueryWithConnection';
export { default as Mutation } from './Mutation';
export { default as PayloadField } from './PayloadField';
export { default as resolveMaybeThunk } from './resolveMaybeThunk';
