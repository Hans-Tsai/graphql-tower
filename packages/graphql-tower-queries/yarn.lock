# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


async-limiter@~1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/async-limiter/-/async-limiter-1.0.0.tgz#78faed8c3d074ab81f22b4e985d79e8738f720f8"

backo2@^1.0.2:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/backo2/-/backo2-1.0.2.tgz#31ab1ac8b129363463e35b3ebb69f4dfcfba7947"

es6-promise@^4.1.1:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/es6-promise/-/es6-promise-4.1.1.tgz#8811e90915d9a0dba36274f0b242dbda78f9c92a"

eventemitter3@^2.0.3:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/eventemitter3/-/eventemitter3-2.0.3.tgz#b5e1079b59fb5e1ba2771c0a993be060a58c99ba"

graphql-subscriptions@^0.5.5:
  version "0.5.5"
  resolved "https://registry.yarnpkg.com/graphql-subscriptions/-/graphql-subscriptions-0.5.5.tgz#f8ad3e094a23128d5065725f9b60020d3293999d"
  dependencies:
    es6-promise "^4.1.1"
    iterall "^1.1.3"

graphql-tower-errors@^3.16.0:
  version "3.16.0"
  resolved "https://registry.yarnpkg.com/graphql-tower-errors/-/graphql-tower-errors-3.16.0.tgz#5dc7b845ef15dc9eb63fa4401f2c324f0cbc8b91"

iterall@^1.1.1, iterall@^1.1.3:
  version "1.1.3"
  resolved "https://registry.yarnpkg.com/iterall/-/iterall-1.1.3.tgz#1cbbff96204056dde6656e2ed2e2226d0e6d72c9"

lodash.assign@^4.2.0:
  version "4.2.0"
  resolved "https://registry.yarnpkg.com/lodash.assign/-/lodash.assign-4.2.0.tgz#0d99f3ccd7a6d261d19bdaeb9245005d285808e7"

lodash.isobject@^3.0.2:
  version "3.0.2"
  resolved "https://registry.yarnpkg.com/lodash.isobject/-/lodash.isobject-3.0.2.tgz#3c8fb8d5b5bf4bf90ae06e14f2a530a4ed935e1d"

lodash.isstring@^4.0.1:
  version "4.0.1"
  resolved "https://registry.yarnpkg.com/lodash.isstring/-/lodash.isstring-4.0.1.tgz#d527dfb5456eca7cc9bb95d5daeaf88ba54a5451"

lodash@^4.17.4:
  version "4.17.4"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.4.tgz#78203a4d1c328ae1d86dca6460e369b57f4055ae"

safe-buffer@~5.1.0:
  version "5.1.1"
  resolved "https://registry.yarnpkg.com/safe-buffer/-/safe-buffer-5.1.1.tgz#893312af69b2123def71f57889001671eeb2c853"

subscriptions-transport-ws@^0.9.1:
  version "0.9.1"
  resolved "https://registry.yarnpkg.com/subscriptions-transport-ws/-/subscriptions-transport-ws-0.9.1.tgz#66488bd5997b6081277a25d6a3015a5aceffed63"
  dependencies:
    backo2 "^1.0.2"
    eventemitter3 "^2.0.3"
    iterall "^1.1.1"
    lodash.assign "^4.2.0"
    lodash.isobject "^3.0.2"
    lodash.isstring "^4.0.1"
    symbol-observable "^1.0.4"
    ws "^3.0.0"

symbol-observable@^1.0.4:
  version "1.0.4"
  resolved "https://registry.yarnpkg.com/symbol-observable/-/symbol-observable-1.0.4.tgz#29bf615d4aa7121bdd898b22d4b3f9bc4e2aa03d"

ultron@~1.1.0:
  version "1.1.1"
  resolved "https://registry.yarnpkg.com/ultron/-/ultron-1.1.1.tgz#9fe1536a10a664a65266a1e3ccf85fd36302bc9c"

ws@^3.0.0, ws@^3.3.2:
  version "3.3.2"
  resolved "https://registry.yarnpkg.com/ws/-/ws-3.3.2.tgz#96c1d08b3fefda1d5c1e33700d3bfaa9be2d5608"
  dependencies:
    async-limiter "~1.0.0"
    safe-buffer "~5.1.0"
    ultron "~1.1.0"
