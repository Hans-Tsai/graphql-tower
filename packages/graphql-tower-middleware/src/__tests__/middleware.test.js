import faker from 'faker';
import { GraphQLInt } from 'graphql';
import { UnauthorizedError } from 'graphql-tower-errors';
import { Query } from 'graphql-tower-queries';
import { authentication } from '../';

describe('middleware', () => {
  it('authentication', async () => {
    const id = faker.random.number();
    const userCard = faker.helpers.userCard();
    const resolve = jest.fn(() => userCard);
    const middleware = jest.fn();

    const QueryAuthentication = class extends Query {
      type = GraphQLInt;
      resolve = resolve;
      middleware = [middleware, authentication];
    };
    const query = new QueryAuthentication();
    expect(query).toMatchSnapshot();

    await expect(query.resolve()).rejects.toEqual(new UnauthorizedError('authorization header is required'));

    await expect(query.resolve({}, {}, faker.random.word())).rejects.toEqual(new UnauthorizedError('authorization header is required'));

    await expect(query.resolve({}, {}, { user: { id } }, {})).resolves.toEqual(userCard);
  });
});
