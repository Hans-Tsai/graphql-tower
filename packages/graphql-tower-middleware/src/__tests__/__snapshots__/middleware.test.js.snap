// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware authentication 1`] = `
QueryAuthentication {
  "_": Object {
    "afterware": Array [],
    "middleware": Array [
      [Function],
      [Function],
    ],
    "parentArgs": undefined,
    "resolve": [Function],
  },
  "args": Object {},
  "resolve": [Function],
  "type": "Int",
}
`;
